packit/base_git.py

```diff
@@ -396,16 +396,46 @@ def set_specfile_content(
         2. The changelog is preserved.
         3. If provided, new version is set.
         4. If provided, new changelog entry is added.
-        5. Release is reset to 1, if the version has changed and the release
-           stayed the same.
+        5. Release is reset to 1 if the version has changed and the release
+           stayed the same or if %autorelease is used only upstream.
 
         Args:
             specfile: specfile to get changes from (we update self.specfile)
             version: version to set in self.specfile
             comment: new comment for the version in %changelog
         """
-        previous_release = self.specfile.release
-        previous_version = self.specfile.expanded_version
+        # determine new release first
+        if self.specfile.has_autorelease:
+            # dist-git spec uses %autorelease, preserve it but prefer the raw value
+            # from upstream spec if it also uses %autorelease
+            new_release = (
+                specfile.raw_release
+                if specfile.has_autorelease
+                else self.specfile.raw_release
+            )
+        else:
+            new_version = specfile.expand(version or specfile.version)
+            if (
+                # if upstream spec uses %autorelease
+                specfile.has_autorelease
```

Comment:
In this case, we cannot update a downsteam non-autorelease to an autorelease. There are two ways around this, either convert it to autorelease using two commits, or inform the packager that they should convert it manually if we don't want to support that.

Reply:
And we are not doing that, in this case we reset the release to `1%{?dist}`. As for converting to `%autorelease` downstream, I plan to propose that for discussion on the next architecture meeting, if we decide we want to support that I will open a separate issue.